tests/subsys/fs/fat_fs_api/src/test_fat_file.c

```diff
@@ -38,7 +38,411 @@ static int test_file_open(void)
 		TC_PRINT("File now exists %s\n", TEST_FILE);
 	}
 
-	return res;
+	return TC_PASS;
+}
+
+static int test_file_open_flags(void)
+{
+	int res;
+	struct fs_file_t fp;
+	char something[] = "Something";
```

Comment:
This should be `const char* something = ` unless you intend to modify the string content (or `const char something[] = ` if you need to do `sizeof()`).